src/components/EventDateFilters.vue

```diff
@@ -1,50 +1,73 @@
 <template>
-  <div class="event-date-filters">
-    <h5 class="event-filters-title">Date of event</h5>
-    <input 
-      type="text" 
-      placeholder="from"
-      :value="startDate"
-      @input="updateSelectedStartDate"
-    />
-    <input 
-      type="text" 
-      placeholder="to"
-      :value="endDate"
-      @input="updateSelectedEndDate"
-    />
+  <div>
+    <h5 v-if="showTitle" class="filter-events-title">Date of event</h5>
+    <div class="filter-events-date-picker">
+      <input ref="startDate" type="text" :value="startDate" />
+      <strong>to</strong>
+      <input ref="endDate" type="text" :value="endDate" />
+    </div>
   </div>
 </template>
 
 <script>
 import { setHash } from 'src/util/url-hash';
+import moment from 'moment';
+
+// sigh
+window.moment = moment;
+
+import Pikaday from 'pikaday';
+
+// e.g., 2017-03-01
+const inputFormat = 'YYYY-MM-DD';
+// e.g., Mar 1, 2017
+const displayFormat = 'MMM D, YYYY'
 
 export default {
   name: 'event-type-filters',
-  props: ['filters'],
-  computed: {
-    startDate() {
-      this.filters.startDate;
-    },
-    endDate() {
-      this.filters.endDate;
-    }
+  props: ['filters', 'showTitle'],
+  data() {
+    const startDate = this.filters.startDate ?
+      moment(this.filters.startDate, inputFormat).format(displayFormat)
+      : null;
+
+    const endDate = this.filters.endDate ?
+      moment(this.filters.endDate, inputFormat).format(displayFormat)
+      : null;
+    
+    return {
+      startDate,
+      endDate
+    };
   },
-  methods: {
-    updateSelectedStartDate(event) {
-      const newDate = event.target.value;
-      // FIXME ha ha
-      if (/\d{4}-\d{2}-\d{2}/.test(newDate)) {
-        setHash({ startDate: newDate });
-      }
-    },
-    updateSelectedEndDate(event) {
-      const newDate = event.target.value;
-      // FIXME ha ha
-      if (/\d{4}-\d{2}-\d{2}/.test(newDate)) {
-        setHash({ endDate: newDate });
-      }
-    }
-  }
+	mounted() {
+    const component = this;
```

Comment:
Yeah definitely, it's a bit duplicative -- OTOH there are some bits of logic that involve the two of them together, e.g., when start date changes, change end date's min date so that it can't be before the new start date, etc. There are other UX improvements that could be made that involve the two dates as a group (doing the opposite when end date changes for example...in fact I should add that :) ).....for now I think for this reason it is probably worth the duplication to allow that precision.